src/App.js

```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
-const App = () => {
-  const [todoList, setTodoList] = React.useState([]);
-  const [isLoading, setIsLoading] = React.useState(true);
 
+const AIRTABLE_API_URL = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
 
-  React.useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []
-          }
-        })
-      }, 2000)
-    })
-      .then((result) => {
-        setTodoList(result.data.todoList);
-        setIsLoading(false);
-      });
+function App() {
+
+  const [todoList, setTodoList] = React.useState([])
+  const [isloading, setIsLoading] = React.useState(true)
+
+  const fetchData = React.useCallback(async () => {
+
+    const options = {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    try {
+      const response = await fetch(AIRTABLE_API_URL, options)
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.title, id: todo.id, }
+      })
+      setTodoList(todos)
+      setIsLoading(false)
+    } catch (error) {
+      console.log(error.message)
+    }
   }, [])
 
-  React.useEffect(() => {
-    if (!isLoading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+  const addTodo = async ({ title }) => {
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+      body: JSON.stringify({
+        fields: {
+          title
+        },
+      }),
+    };
+    try {
+      const response = await fetch(AIRTABLE_API_URL, options);
+      if (!response.ok) {
+        throw new Error(`Something went wrong: ${response.status}`);
+      }
+
+      const airtableData = await response.json();
+      setTodoList((prevTodoList) => [
+        ...prevTodoList,
+        {
+          id: airtableData.id,
+          title: airtableData.fields.title,
+        },
+      ]);
+    } catch (error) {
+      console.log(error);
     }
-  }, [todoList, isLoading]);
+  };
+
 
-  const removeTodo = (id) => {
-    setTodoList(prevTodoList => {
-      return prevTodoList.filter(
-        (todo) => todo.id !== id
+  const removeTodo = async (id) => {
+    try {
+      const response = await fetch(`${AIRTABLE_API_URL}/${id}`, {
```

Comment:
Depending on what the return object is from your delete call, you can potentially reuse your state setter function for your `delete()` and `addTodo()`